### PR TITLE
fix: modify function updateAssociatedProfile return errors

### PR DESCRIPTION
**Pull request by gy95**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## What is the current behavior?

1. if profile not exist, after adding profile to cache map, update profile again
2. if update profile failed, return nil

## What is the new behavior?
1. if profile not exist, after adding profile to cache map, dont need to update profile again
2. if update profile failed, return err just like add profile failed

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gy95** [comment]
how to fix these check errors?

**Comment by cloudxxx8** [comment]
@gy95 you need to follow the conventional commits and commit message in the guide:
https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-ConventionalCommits

the commit types are defined here:
https://github.com/edgexfoundry/device-sdk-go/blob/master/.github/semantic.yml

File: internal/application/callback.go [hunk 1]
```diff
@@ -256,10 +256,12 @@ func updateAssociatedProfile(profileName string, dic *di.Container) errors.EdgeX
 			errMsg := fmt.Sprintf("failed to add profile %s", profileName)
 			return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
 		}
+		return nil
 	}
 	err = cache.Profiles().Update(dtos.ToDeviceProfileModel(res.Profile))
 	if err != nil {
 		lc.Warnf("failed to to update profile %s in cache, using the original one", profileName)
+		return err
```
Comment (cloudxxx8) [inline internal/application/callback.go:264]: please use the EdgeX error type to wrap the error here

```
errMsg := fmt.Sprintf(failed to to update profile %s in cache, using the original one", profileName)
return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
```

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/application/callback.go (hunk 1 above)
Comment (gy95) [reply internal/application/callback.go:264]: ok

**Comment by cloudxxx8** [comment]
@gy95 please rebase the branch to the latest version

**Review by cloudxxx8 — APPROVED** [review]
the change looks good to me, but you need to pull the latest changes from master and rebase to your branch before pushing.  it's not based on the latest master branch now.

**Pull request merged**